timings/timings.go

```diff
@@ -0,0 +1,98 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package timings
+
+import (
+	"time"
+)
+
+var timeNow = func() time.Time {
+	return time.Now()
+}
+
+// Timings represents a tree of Timing measurements for a single execution of measured activity.
+// A Timings tree object should be created at the beginning of the activity,
+// followed by starting at least one Timing measurement, and then saved at the end of the activity.
+//
+// Calling Start on the Timings objects creates a Timing and starts new
+// performance measurement. Measurement needs to be finished by calling Stop
+// function on the Timing object.
+// Nested measurements may be collected by calling Start on Timing objects. Similar
+// to the above, nested measurements need to be finished by calling Stop on them.
+//
+// Typical usagage:
+//   timingsTree := timings.New(map[string]string{"tag": "tag-value"})
```

Comment:
Thanks for the suggestions, done. I think this doc comment will need updating again after the first PR using this API lands, as it will grow some helpers (such as a helper to create Timings from a task with proper tags defined automatically).